### DISCUSSION TITLE
Bug Report: Flip Selection Issues

### DISCUSSION THREAD

**Discussion by DinoAwakening** [keiwando/toonsquid#295] [Q&A] [answered]
https://user-images.githubusercontent.com/120270913/236706147-c9f31b18-bc10-47e7-a38e-86f00a9b051c.mp4

**Comment by keiwando** [comment]
Thanks for the video! I'll be looking into this.

**Reply by keiwando** [reply]
@DinoAwakening I have not been able to reproduce this issue. Does this happen in all projects for you when you flip a selection within a pixel layer?

**Reply by DinoAwakening** [reply]
Yes it does D:

[iPad Air 5, iOS 16.4.1(a)]

**Reply by keiwando** [reply]
I have also tried it on that iPad model and unfortunately couldn't reproduce it there either. Are there any obvious customized settings (both within ToonSquid but also in your iPad itself) or other non-standard aspects of your setup that you can think of? Maybe a setting somewhere is interfering with this behaviour somehow.

**Reply by DinoAwakening** [reply]
I’m not sure what could it be. It’s not that annoying anyway... Time will bring the answer eventually XD

**Comment by Cat-Butler** [comment]
What exactly is the bug in this case?
Is it that when selecting and flipping a selection, the selection changes to the entire layer contents rather than retaining the selected content? Or am I on the wrong track completely?

If that is the problem, I get that too, on an iPad Pro 10.5", and an iPad Mini 5, both iPadOS 16.4.1(a).
Assuming I'm not mistaken about the bug, my reproduction process:

1. Start a new project (1920x1080 default).
2. Use a raster/pixel brush and draw something (naturally, this creates a new pixel Drawing layer).
3. Change to the Selection tool and select an area (freehand) of the drawn content.
4. Change to the Transform tool and click a flip icon to flip the selected area.

Result:
Selected area successfully flips, but the Transform box immediately changes to all the pixels on the layer (instead of retaining just the selected/flipped content). Note: selection area highlight also disappears.
Apparently still in pixel edit mode (icon highlighted), but acts like you've turned it off.

**Reply by DinoAwakening** [reply]
You got that right. Although, the resolution is irrelevant D:

**Reply by keiwando** [reply]
Just for reference: This is the (correct) behavior I'm seeing on an iPad Air 5 running 16.4.1(a) and I still have no idea why you can reliably reproduce this issue but I can't.

https://github.com/keiwando/toonsquid/assets/12989570/8b7e47a9-749b-40bf-a58b-c204ccd54694

**Comment by Cat-Butler** [comment]
Don't know if this will help, but here are my settings (bug reproduces consistently):

## Settings

### General

#### General 

Play audio while scrubbing the timeline: ON
Quick layer delete button: Delete drawing layer
Layer pivot editing enabled: ON
Follow the camera layer during playback: ON
Reset the frame after pausing playback: ON
Forward / Backward buttons should move between: Frames

#### Drawings 

Default drawing length (in frames): 1
Start frame of new drawings: Current frame

#### Keyframes

Show keyframes for: The selected layer 
Automatically add keyframes: ON 

#### Animation layer merging 

Drawing layer merge behaviour: Separate 

#### Copy & Paste 

Merge down pixel and vector layers after pasting: OFF

### Tools 

#### Brush 

Minimum tilt: 10%
Maximum tilt: 75%
Cursor enabled (brush tool): OFF 
Cursor enabled (ereader tool): ON 
Cursor enabled (smudge tool): OFF 

#### Fill

Source: All layers 
Close gaps (Pixel Layer): ON 
Gap size: 10px

### Appearance 

#### Appearance 

Theme: Dark 
Show a background grid: OFF 
Selection mask visibility: 41%

### Gestures 

#### General 

Finger drawing: OFF 

#### History 

Two finger tap to undo: ON 
Three finger tap to redo: ON 

#### Pipette 

Long press for pipette (with finger): ON 
Long press for pipette (with pencil): OFF 
Long press for pipette enabled in the fill tool: OFF 

#### Page flip 

Three finger swipe (up / down) to flick through frames: ON 
Number of frames before and after to flip through: 5
Swipe sensitivity: 50%
Invert swipe direction: OFF 
Reset to original frame after swipe: ON 

#### Text

Drag in text tool: Create selection 

### Backup 

Backup reminders enabled: ON 
Reminder interval in minutes: 20
Number of backups to keep: 2

https://github.com/keiwando/toonsquid/assets/25494772/9cbbe303-5db6-4c3b-b9f4-c8d2548f2f14

**Reply by keiwando** [reply]
Thank you very much for the detailed list of settings. I tried it with all of those but was unfortunately still unable to reproduce the issue.

**Reply by DinoAwakening** [reply, marked as answer]
> Thank you very much for the detailed list of settings. I tried it with all of those but was unfortunately still unable to reproduce the issue.

You can just forget about it. 1.9 somehow fixed the problem :D
